pkg/controller/gitopscluster/gitopscluster_controller.go

```diff
@@ -767,3 +769,43 @@ func (r *ReconcileGitOpsCluster) CreateManagedClusterSecretInArgo(argoNamespace
 
 	return nil
 }
+
+func unionSecretData(newSecret, existingSecret *v1.Secret) *v1.Secret {
+	newLabels := newSecret.GetLabels()
+	existingLabels := existingSecret.GetLabels()
+
+	// union of labels
+	for key, val := range existingLabels {
+		newLabels[key] = val
```

Comment:
Just to give some more context, the newLabels map itself consists of only 4 labels:
```
"argocd.argoproj.io/secret-type":                 "cluster",
"apps.open-cluster-management.io/acm-cluster":    "true",
"apps.open-cluster-management.io/cluster-name":   labels["apps.open-cluster-management.io/cluster-name"],
"apps.open-cluster-management.io/cluster-server": labels["apps.open-cluster-management.io/cluster-server"],
```
Of which the values shouldn't be different from the existingLabels values from the caller function.

Reply:
I think Mike's comment makes sense.  The 4 labels Jonathan listed should be from the new secret anyway.  In particular, the `cluster-name` and `cluster-server` could change in case the managed cluster is re-imported.

So I think it would be safer to do this union. 
1. for the 4 labels, always apply the values from the new secret.
2. merge other labels from the existing secret to the new secret.

Reply:
Managed to cover both cases via only adding new labels that are not already in the newSecret labels map. This allows for both the 4 listed labels to already be in the map and for other labels to be merged.